pyfr/plugins/integrate.py

```diff
@@ -52,22 +52,45 @@ def __init__(self, intg, cfgsect, suffix=None):
             # Open
             self.outf = init_csv(self.cfg, cfgsect, ','.join(header))
 
+        # Whether to use a quadrature different to that of the solution points
+        quad = self.cfg.hasopt(cfgsect, 'quad-deg')
+
         # Prepare the per element-type info list
-        self.eleinfo = []
+        self.eleinfo = eleinfo = []
         for (ename, eles), (eset, emask) in zip(system.ele_map.items(), rinfo):
-            # Locations of each solution point
-            ploc = eles.ploc_at_np('upts')[..., eset]
-            ploc = ploc.swapaxes(0, 1)
+            # Obtain quadrature info
+            rname = self.cfg.get(f'solver-elements-{ename}', 'soln-pts')
 
-            # Jacobian determinants
-            rcpdjacs = eles.rcpdjac_at_np('upts')[:, eset]
+            # Raise error with edge cases
+            if not quad and (self.cfg.hasopt(cfgsect, f'quad-pts-{ename}')
+                             or self.cfg.hasopt(cfgsect, f'quad-deg-{ename}')):
+                raise ValueError('Option "quad-deg" is not found')
 
-            # Quadature weights
-            rname = self.cfg.get(f'solver-elements-{ename}', 'soln-pts')
-            wts = get_quadrule(ename, rname, eles.nupts).wts
+            if quad:
+                # Quadrature rule (default to that of the solution points)
+                qrule = self.cfg.get(cfgsect, f'quad-pts-{ename}', rname)
+
+                # Quadrature rule degree
+                qdeg = self.cfg.getint(cfgsect, 'quad-deg')
+                if self.cfg.hasopt(cfgsect, f'quad-deg-{ename}'):
```

Comment:
Does this mean that I need to specify quad-deg even if I explicitly specify the rule degree for each element type in my simulation?

Reply:
Yes, although this can be modified. What would happen if the user does not specify the quadrature degree of an element type? Do we want to force the user to set the quadrature degree of all element types?

Reply:
We could maybe change those lines to

```python
try:
    qdeg = self.cfg.getint(cfgsect, f'quad-deg-{ename}')
except:
    qdeg = self.cfg.getint(cfgsect, f'quad-deg')
```

This would give preference to `quad-deg-{ename}` over `quad-deg`

Reply:
Ok. I've uploaded the new solution

1) Try with `quad-deg-{ename}`.
2) Except 1. try with `quad-deg`.
3) Except 2. use the quadrature of the solution points.

Let me know what you think!